tests/unit/components/NcRichText/autoLink.spec.js

```diff
@@ -0,0 +1,159 @@
+ *
+ */
+
+import { expect, describe, it, jest } from '@jest/globals'
+import { getRoute } from '../../../../src/components/NcRichText/autolink.js'
+import VueRouter from 'vue-router'
+import { getBaseUrl, getRootUrl } from '@nextcloud/router'
+
+jest.mock('@nextcloud/router')
+
+describe('autoLink', () => {
+	describe('getRoute', () => {
+		describe.each([
+			['an absolute link', 'https://cloud.ltd'],
+			['a relative link', ''],
+		])('for %s', (_, origin) => {
+			describe.each([
+				['with base /nextcloud', '/nextcloud'],
+				['on server root', ''],
+			])('%s', (_, root) => {
+				describe.each([
+					['with', '/index.php'],
+					['without', ''],
+				])('%s /index.php in link', (_, indexPhp) => {
+					const linkBase = origin + root + indexPhp
+					beforeAll(() => {
+						getBaseUrl.mockReturnValue(`https://cloud.ltd${root}`)
+						getRootUrl.mockReturnValue(root)
+					})
+
+					describe.each([
+						['with', '/index.php'],
+						['without', ''],
+					])('%s /index.php in router base', (_, indexPhpInRouterBase) => {
+						const routerBase = `${root}${indexPhpInRouterBase}`
+
+						it.each([
+							[`${linkBase}/apps/test/foo`, '/foo'],
+							[`${linkBase}/apps/test/foo/`, '/foo/'],
+							[`${linkBase}/apps/test/bar/1`, '/bar/1'],
+
+							['https://external.ltd/nextcloud/index.php/apps/test/foo', null], // Different origin
+							['https://cloud.ltd/external/index.php/apps/test/foo/', null], // Different base
+							['https://cloud.ltd/nextcloud/index.php/apps/not-router-base/', null], // Different router base
+							['https://cloud.ltd/nextcloud/apps/test/baz', null], // No matching route
+						])('should get route %s => %s', (link, expectedRoute) => {
+							const routerTest = new VueRouter({
+								mode: 'history',
+								base: `${routerBase}/apps/test`,
+								routes: [
+									{ path: '/foo', name: 'foo' },
+									{ path: '/bar/:param', name: 'bar' },
+								],
+							})
+							expect(getRoute(routerTest, link)).toBe(expectedRoute)
+						})
+
+						it.each([
+							[`${linkBase}/apps/files/`, '/files'],
+							[`${linkBase}/apps/files/favorites/1`, '/favorites/1'],
+							[`${linkBase}/apps/files/files/1?fileid=2#c`, '/files/1?fileid=2#c'], // With query and hash
+							[`${linkBase}/apps/files/files/1?dir=server/lib/index.php#c`, '/files/1?dir=server%2Flib%2Findex.php#c'], // With index.php in query
+
+						])('should get route for Files: %s => %s', (link, expectedRoute) => {
+							const routerFiles = new VueRouter({
+								mode: 'history',
+								base: `${routerBase}/apps/files`,
+								routes: [
+									{ path: '/', name: 'root', redirect: '/files' },
+									{ path: '/:view/:fileid(\\d+)?', name: 'fileslist' },
+								],
+							})
+
+							expect(getRoute(routerFiles, link)).toBe(expectedRoute)
+						})
+
+						it.each([
+							[`${linkBase}/apps/spreed?callTo=alice`, '/apps/spreed?callTo=alice'],
+							[`${linkBase}/call/abc123ef`, '/call/abc123ef'],
```

Comment:
I see it for Files tests, but we're also using hash to navigate to specific message in the chat
```suggestion
							[`${linkBase}/call/abc123ef`, '/call/abc123ef'],
							[`${linkBase}/call/abc123ef#message_123`, '/call/abc123ef#message_123'],
```

Reply:
Added